src/index.ts

```diff
@@ -231,14 +245,24 @@ export function findByDevice(device: Zh.Device) {
     return definition;
 }
 
-export function findDefinition(device: Zh.Device): Definition {
+export function findDefinition(device: Zh.Device, generateForUnknown: boolean = false): Definition {
     if (!device) {
         return null;
     }
 
     const candidates = getFromLookup(device.modelID);
     if (!candidates) {
-        return null;
+        if (!generateForUnknown || device.type === 'Coordinator') {
+            return null;
+        }
+
+        const definition = generateDefinition(device);
+        if (!definition) {
+            return null
```

Comment:
Let's never return `null` here, always return a definition and make sure the ones generated have `generated = true` on it

Reply:
Done, always returning something with `generated = true`.